### PR TITLE
Bluetooth: Controller: Fix ticker to prefer ticker node started

### PR DESCRIPTION
**Pull request by cvinayak**
Fix ticker implementation to prefer to keep ticker node stated in case of race condition to start and stop.

This fixes assertion in lll.c due to prepare pipeline overflow due to preempt ticker being stopped under race conditions of ticker start/stop sequence.
The assertion at: https://github.com/zephyrproject-rtos/zephyr/blob/c61502ccbed19aa3eec0d7d4c42a29284ab34e76/subsys/bluetooth/controller/ll_sw/nordic/lll/lll.c#L699

This can be reproduced running `samples/bluetooth/periodic_sync` with it synchronized to `samples/bluetooth/periodic_adv` sample for a long duration.

Regression introduced in commit ed92ea9

The change required was identified during code inspection related to analysis of https://devzone.nordicsemi.com/f/nordic-q-a/103127/connectless-cte-tx-rx-problem

### CODE REVIEW

File: subsys/bluetooth/controller/Kconfig.ll_sw_split [hunk 1]
```diff
@@ -1036,6 +1038,21 @@ config BT_TICKER_SLOT_AGNOSTIC
 	  reservations and collision handling, and operates as a simple
 	  multi-instance programmable timer.
 
+config BT_TICKER_PREFER_START_BEFORE_STOP
```
Comment (Thalley) [inline subsys/bluetooth/controller/Kconfig.ll_sw_split:1041]: Is this being tested with both `y` and `n` in CI?

Comment (cvinayak) [reply subsys/bluetooth/controller/Kconfig.ll_sw_split:1041]: Apparently there is no coverage for `BT_TICKER_SLOT_AGNOSTIC`.

@mtpr-ot should I just add `BT_TICKER_SLOT_AGNOSTIC` and `BT_TICKER_PREFER_START_BEFORE_STOP` as a build only test in `tests/bluetooth/init` samples?

Comment (cvinayak) [reply subsys/bluetooth/controller/Kconfig.ll_sw_split:1041]: Added a prj_ctlr_ticker.conf to cover the changes in this PR.

Comment (mtpr-ot) [reply subsys/bluetooth/controller/Kconfig.ll_sw_split:1041]: > Apparently there is no coverage for `BT_TICKER_SLOT_AGNOSTIC`.
> 
> @mtpr-ot should I just add `BT_TICKER_SLOT_AGNOSTIC` and `BT_TICKER_PREFER_START_BEFORE_STOP` as a build only test in `tests/bluetooth/init` samples?

That would be very nice, thanks.

**Review by mtpr-ot — DISMISSED** [review]
LGTM

**Review by Thalley — DISMISSED** [review]

File: subsys/bluetooth/controller/Kconfig.ll_sw_split (hunk 1 above)
Comment (cvinayak) [reply subsys/bluetooth/controller/Kconfig.ll_sw_split:1041]: Added and CI passing now

**Review by Thalley — APPROVED** [review]

**Review by mtpr-ot — APPROVED** [review]

**Pull request merged**